chamber/utils/__init__.py

```diff
@@ -43,15 +47,24 @@ def keep_spacing(value, autoescape=True):
     return mark_safe(value.replace('  ', ' &nbsp;').replace('\n', '<br />'))
 
 
-def call_method_with_unknown_input(method, **fun_kwargs):
-    method_kwargs_names = inspect.getargspec(method)[0][1:]
+def call_function_with_unknown_input(function, **kwargs):
+    """
+    Call function and use kwargs from input if function requires them.
+    :param function: function to call.
+    :param kwargs: function input kwargs or extra kwargs which will not be used.
+    :return: function result or raised InvalidFunctionArguments exception.
+    """
+    function_kwargs_names = set(inspect.signature(function).parameters.keys())
 
-    method_kwargs = {arg_name: fun_kwargs[arg_name] for arg_name in method_kwargs_names if arg_name in fun_kwargs}
+    required_function_kwargs_names = set(
+        k for k, v in inspect.signature(function).parameters.items() if v.default is v.empty
+    )
 
-    if len(method_kwargs_names) == len(method_kwargs):
-        return method(**method_kwargs)
+    function_kwargs = {k: v for k, v in kwargs.items() if k in function_kwargs_names}
+    if required_function_kwargs_names <= set(function_kwargs.keys()):
+        return function(**function_kwargs)
     else:
-        raise RuntimeError('Invalid method parameters')
+        raise InvalidFunctionArguments('Function {} arguments has not subset of fun kwargs'.format(function))
```

Comment:
Maybe a better message would be `Function {} missing required arguments {}` and include the missing required arguments?